### PR TITLE
Added : #define CLIENT_MIN_BUILD & CLIENT_MAX_BUILD & BUILD_VERSION.

### PR DESCRIPTION
**Pull request by schnek**

### CODE REVIEW

File: configs/world.conf [hunk 1]
```diff
@@ -196,7 +200,8 @@
 #        Default: 0 (off)
 #
 
-<Server PlayerLimit          = "100"
+<Server CacheVersion         = "12340"
```
Comment (Zyres) [inline configs/world.conf]: CacheVersion is equal ro GameBuild.... why do we need 2 different configs with the same values?

File: src/logonserver/Auth/AuthSocket.cpp [hunk 2]
```diff
@@ -592,7 +592,7 @@ void AuthSocket::HandleReconnectChallenge()
     readBuffer.Read(&m_challenge, full_size + 4);
 
     // Check client build.
-    if (m_challenge.build > sLogonServer.clientMaxBuild || m_challenge.build < sLogonServer.clientMinBuild)
+    if (m_challenge.build)
```
Comment (Zyres) [inline src/logonserver/Auth/AuthSocket.cpp]: So if the challenge.build is greater 0 the client received an error msg?

File: src/world/Map/TerrainMgr.cpp [hunk 3]
```diff
@@ -405,14 +405,13 @@ void TileMap::Load(char* filename)
         return;
     }
 
-    if (header.buildMagic != 12340)  //wow version
+    if (header.buildMagic != sWorld.CacheVersion) // wow version
```
Comment (Zyres) [inline src/world/Map/TerrainMgr.cpp]: This value should be a hardcoded value. It depends on our extractors. the TerrainMgr maybe works different on another game version....

File: src/logonserver/Server/Master.cpp [hunk 4]
```diff
@@ -142,9 +142,6 @@ void LogonServer::Run(int argc, char** argv)
     std::string host = Config.MainConfig.GetStringDefault("Listen", "Host", "0.0.0.0");
     std::string shost = Config.MainConfig.GetStringDefault("Listen", "ISHost", host.c_str());
 
-    clientMinBuild = 12340;
-    clientMaxBuild = 12340;
```
Comment (Zyres) [inline src/logonserver/Server/Master.cpp]: We need these members for further development (especially multiversion support).

File: src/world/Map/TerrainMgr.cpp (hunk 3 above)
Comment (schnek) [inline src/world/Map/TerrainMgr.cpp]: can make the macro in MapManagementGlobals.hpp

ADD `#define GAME_BUILD 12340 // wow version`

File: src/logonserver/Server/Master.cpp (hunk 4 above)
Comment (schnek) [inline src/logonserver/Server/Master.cpp]: then it is better to return them to .conf (original ascent)

```
##################################################
# Accepted Build Range Setup
#
# These two directives set up which clients will be
# Allowed to authenticate with the realm list.
#
# Set these to the same builds that the server was
# Compiled for.
#
# As of the last update, version 3.3.5a was build 12340.
#
##################################################
<Client MinBuild = "12340" MaxBuild = "12340">
```

File: configs/world.conf (hunk 1 above)
Comment (schnek) [inline configs/world.conf]: https://github.com/AscEmu/AscEmu/pull/411#discussion_r101307692

File: src/logonserver/Auth/AuthSocket.cpp (hunk 2 above)
Comment (schnek) [inline src/logonserver/Auth/AuthSocket.cpp]: I did not get errors, but on logic, you are right

File: src/logonserver/Server/Master.cpp (hunk 4 above)
Comment (Zyres) [inline src/logonserver/Server/Master.cpp]: A config is useless... there is nothing the user is allowed to configure. We decide which versions are supported based on the programm.

**Comment by Zyres** [comment]
Sorry but I have to close your PR. the changes in the logonserver proj. is not accepted since the logonserver is a github module. I know we copied the logonserver temporarily to the branch but it will be moved back to the github module soon.

Thank you for your contribution.

**Pull request closed without merging**